### PR TITLE
JAMES-2272 Prevent forgetting Location header upon webadmin task generation

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/protocols/webadmin/webadmin-cassandra/src/test/java/org/apache/james/webadmin/routes/CassandraMigrationRoutesTest.java [hunk 1]
```diff
@@ -232,6 +232,18 @@ public void postShouldNotUpdateVersionWhenCurrentVersionIsNewerThan() {
...
 
+    @Test
+    public void postShouldPositionLocationHeader() {
+        when(schemaVersionDAO.getCurrentSchemaVersion()).thenReturn(Mono.just(Optional.of(CURRENT_VERSION)));
+
+        given()
+            .body(String.valueOf(OLDER_VERSION.getValue()))
+        .when()
+            .post("/upgrade")
+            .then()
```
Comment (Arsnael) [inline server/protocols/webadmin/webadmin-cassandra/src/test/java/org/apache/james/webadmin/routes/CassandraMigrationRoutesTest.java]: indent

File: server/protocols/webadmin/webadmin-cassandra/src/test/java/org/apache/james/webadmin/routes/CassandraMigrationRoutesTest.java [hunk 2]
```diff
@@ -232,6 +232,18 @@ public void postShouldNotUpdateVersionWhenCurrentVersionIsNewerThan() {
...
+            .header("Location", notNullValue());
```
Comment (trantienduchn) [inline server/protocols/webadmin/webadmin-cassandra/src/test/java/org/apache/james/webadmin/routes/CassandraMigrationRoutesTest.java]: what is the purpose of this header?

Comment (chibenwa) [reply server/protocols/webadmin/webadmin-cassandra/src/test/java/org/apache/james/webadmin/routes/CassandraMigrationRoutesTest.java]: It indicates that a resource related to this call had been created on another endpoint

**Comment by chibenwa** [comment]
`MemorySpamAssassinContractTest.imapMovesToSpamMailboxShouldBeConsideredAsSpam » ConditionTimeout`

test this please

**Comment by Arsnael** [comment]
```
[ERROR]   MemorySpamAssassinContractTest.expungingSpamMessageShouldNotImpactSpamAssassinState » ConditionTimeout
```
test this please

File: server/protocols/webadmin/webadmin-cassandra/src/main/java/org/apache/james/webadmin/routes/CassandraMigrationRoutes.java [hunk 3]
```diff
@@ -144,7 +144,7 @@ public Object upgradeToVersion(Request request, Response response) {
         CassandraVersionRequest cassandraVersionRequest = CassandraVersionRequest.parse(request.body());
         Task migration = cassandraMigrationService.upgradeToVersion(cassandraVersionRequest.getValue());
         TaskId taskId = taskManager.submit(migration);
-        return TaskIdDto.from(taskId);
+        return TaskIdDto.respond(response, taskId);
```
Comment (mbaechler) [inline server/protocols/webadmin/webadmin-cassandra/src/main/java/org/apache/james/webadmin/routes/CassandraMigrationRoutes.java]: why a DTO is supposed to have a method to interact with a response?
It doesn't look like a DTO anymore

Comment (chibenwa) [reply server/protocols/webadmin/webadmin-cassandra/src/main/java/org/apache/james/webadmin/routes/CassandraMigrationRoutes.java]: See https://github.com/linagora/james-project/pull/2990/commits/c1f4e0cba9f0fb66ef46a02700eb2eadb3e53e3d

I will remove the part related to your remark from this work.

**Comment by Arsnael** [comment]
Merged

**Pull request closed without merging**